Week1/queries.js

```diff
@@ -0,0 +1,147 @@
+const sqlCreateWorldDB = `CREATE DATABASE IF NOT EXISTS world;`;
+const sqlUseWorldDB = `USE world;`;
+const sqlCreateCountry = `
+CREATE TABLE country (
+  Code VARCHAR(3) NOT NULL,
+  Name VARCHAR(50) NOT NULL,
+  Continent ENUM('Asia', 'Europe', 'North America', 'Africa', 'Oceania', 'Antarctica', 'South America') NOT NULL,
+  Region VARCHAR(50) NULL,
+  SurfaceArea FLOAT(10,2) NULL DEFAULT 0.00,
+  IndepYear INT NULL,
+  Population INT NULL DEFAULT 0,
+  LifeExpectancy DECIMAL(3,1) NULL,
+  GNP FLOAT(10,2) NULL,
+  GNPOld FLOAT(10,2) NULL,
+  LocalName VARCHAR(50) NOT NULL,
+  GovernmentForm VARCHAR(50) NULL,
+  HeadOfState VARCHAR(50) NULL,
+  Capital INT NULL,
```

Comment:
Actually I forgot to add foreign key of capital referencing to city id. That is why i am using capital as `INT`. Added country_fk foreign key

```
ALTER TABLE country
ADD CONSTRAINT country_fk
  FOREIGN KEY (Capital)
  REFERENCES city (ID);
```

Reply:
So first we are discussing on an other level than what is asked on the homework here :)
I totally get what you are trying to do but I'm not a big fan of this foreign key. It will make row manipulations difficult even almost impossible. In fact your sample is working only because you added the constraint after adding the data. But if you try to add a new row now you will see that it becomes hard. Because to insert a country you need a city and to insert a city you need a country :)
If you want to go that way, I think the best would be to extract the CountryCode into a third table. Then add a foreign key on this table in both Country and City.

Reply:
You are right that the relationship between country and city tables most likely cause deadlocks with new inserts. At least we must have already inserted all capitals with all countries before using this constraint. The best solution will be -as you mentioned- making a CapitalOfCountries table between two tables having country_code and city_id referencing country and city tables respectively.
This mutual foreign key references also preventing from dropping the tables without dropping the foreign key 😁